pyclean/modern.py

```diff
@@ -90,6 +90,50 @@ def configure(self, args):
 Runner = CleanupRunner()
 
 
+def should_ignore(path, ignore_patterns):
+    """
+    Check if a path should be ignored based on ignore patterns.
+
+    Patterns can be:
+    - Simple names like 'bar': matches any directory with that name
+    - Paths like 'foo/bar': matches 'bar' directory inside 'foo' directory
+      and also ignores everything inside that directory
+
+    Args:
+        path: Path object to check
+        ignore_patterns: List of ignore patterns
+
+    Returns:
+        True if the path should be ignored, False otherwise
+    """
+    if ignore_patterns is None:
+        return False
+    if not ignore_patterns:
+        return False
+
+    for pattern in ignore_patterns:
+        if '/' in pattern:
```

Comment:
Will that also work with Windows-style paths on Windows? A user running PyClean on a Windows machine might enter `pyclean C:\ -v -i C:\WINDOWS\SYSTEM32` in cmd or PowerShell.

Should we add tests that run cases like that on Windows only?